demo/DemoArticles/update.js

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env node
+/* eslint-disable require-jsdoc */
+
+const fs = require('fs')
+const request = require('request')
+const indexJSON = require('./articles.json')
+const ArticleRef = require('./ArticleRef.js').ArticleRef
+const ArticleRefSourceType = require('./ArticleRef.js').ArticleRefSourceType
+const flattenArrayOfArrays = require('../DemoUtilities.js').flattenArrayOfArrays
+
+const DATA_PATH = './data/'
+
+const articleRefs = flattenArrayOfArrays(
+  indexJSON.map(articleData => [
+    new ArticleRef(
+      articleData.lang,
+      articleData.title,
+      articleData.revision,
+      ArticleRefSourceType.mobileView
```

Comment:
> The client of ArticleRef can just supply this...

True, but there's a big benefit to more complete ArticleRef modeling - Client code only has to specify the sourceType _once_ when creating an ArticleRef, and later calls to its methods which are variant on sourceType (sections, url, filename, display name, etc) _just work_ without having to maintain external sourceType state. 

This made it much simpler to use them (via functional fun) in multiple places in very different ways - i.e. by the update script to fetch, by the demo to construct menu items and also to present sections.